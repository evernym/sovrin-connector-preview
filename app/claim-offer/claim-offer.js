// @flow
import React, { PureComponent } from 'react'
import { StyleSheet, Platform, FlatList, View } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import {
  claimOfferShown,
  acceptClaimOffer,
  claimOfferRejected,
  claimOfferIgnored,
  claimOfferShowStart,
  resetClaimRequestStatus,
} from './claim-offer-store'
import {
  Container,
  CustomView,
  CustomButton,
  CustomText,
  Icon,
  ConnectionTheme,
  ClaimProofHeader,
  Separator,
  FooterActions,
  CustomSafeAreaView,
  headerStyles,
} from '../components'
import { homeRoute } from '../common/'
import {
  OFFSET_2X,
  OFFSET_1X,
  color,
  OFFSET_3X,
  OFFSET_5X,
} from '../common/styles'
import type {
  AdditionalDataPayload,
  Attribute,
} from '../push-notification/type-push-notification'
import type {
  ClaimOfferProps,
  ClaimOfferPayload,
  ClaimOfferAttributeListProps,
  ClaimOfferState,
} from './type-claim-offer'
import { CLAIM_REQUEST_STATUS } from './type-claim-offer'
import type { Store } from '../store/type-store'
import ClaimRequestModal from './claim-request-modal'
import {
  getConnectionLogoUrl,
  getConnectionTheme,
} from '../store/store-selector'

import type { ReactNavigation } from '../common/type-common'
import { updateStatusBarTheme } from '../../app/store/connections-store'
import { CustomModal } from '../components'
import PaymentFailureModal from '../wallet/payment-failure-modal'

class ClaimOfferAttributeList extends PureComponent<
  ClaimOfferAttributeListProps,
  void
> {
  keyExtractor = ({ label }: Attribute, index: number) => `${label}${index}`

  renderItem = ({ item }: { item: Attribute }) => {
    return (
      <CustomView fifth row horizontalSpace doubleVerticalSpace>
        <CustomView fifth right style={[styles.attributeListLabel]}>
          <CustomText
            h7
            uppercase
            semiBold
            bg="tertiary"
            transparentBg
            style={[styles.attributeListLabelText]}
          >
            {item.label}
          </CustomText>
        </CustomView>
        <CustomView fifth left style={[styles.attributeListValue]}>
          <CustomText h6 demiBold bg="tertiary" transparentBg>
            {item.data}
          </CustomText>
        </CustomView>
      </CustomView>
    )
  }

  render() {
    const attributes: Array<Attribute> = this.props.list
    return (
      <Container fifth style={[styles.claimOfferData]}>
        <FlatList
          data={attributes}
          keyExtractor={this.keyExtractor}
          ItemSeparatorComponent={Separator}
          ListFooterComponent={Separator}
          renderItem={this.renderItem}
        />
      </Container>
    )
  }
}

export class ClaimOffer extends PureComponent<
  ClaimOfferProps,
  ClaimOfferState
> {
  constructor(props: ClaimOfferProps) {
    super(props)
    if (props.navigation.state) {
      props.claimOfferShowStart(props.navigation.state.params.uid)
    }
  }

  state = {
    disableAcceptButton: false,
    insufficientBalanceModalHidden: false,
  }

  close = () => {
    this.props.navigation.goBack()
  }

  hideInsufficientBalanceModal = () => {
    // the reason why we can't just use close method is our CustomModal
    // somehow on some Android devices, if we open claim offer screen again
    // modal component flashes for a second and then goes away
    // however, props to the component are updated ones and does not satisfy
    // condition to show modal, but Modal still shows up
    // We have tried using this format as well isValid ? <Modal /> : null
    // However, in this format, Modal freezes the screen with Modal's background
    // and user can't tap anywhere on screen and app seems to be frozen
    // then only workaround is to kill the app and run again
    // So we can't go back to ternary based null rendering as well
    this.props.resetClaimRequestStatus(this.props.uid)
  }

  onIgnore = () => {
    this.props.claimOfferIgnored(this.props.uid)
    this.close()
  }

  onReject = () => {
    this.props.claimOfferRejected(this.props.uid)
    this.close()
  }

  onAccept = () => {
    this.setState({
      disableAcceptButton: true,
    })
    this.props.acceptClaimOffer(this.props.uid)
  }

  onInsufficientModalHide = () => {
    this.setState({
      insufficientBalanceModalHidden: true,
    })
  }

  componentDidMount() {
    // update store that offer is shown to user
    this.props.claimOfferShown(this.props.uid)
    this.props.updateStatusBarTheme(this.props.claimThemePrimary)
  }

  componentDidUpdate(prevProps: ClaimOfferProps, prevState: ClaimOfferState) {
    if (
      this.state.insufficientBalanceModalHidden !==
        prevState.insufficientBalanceModalHidden &&
      this.state.insufficientBalanceModalHidden
    ) {
      // once we are sure that insufficient balance modal was properly hidden
      // we can go ahead and close claim offer screen as well
      this.close()
    }
  }

  render() {
    const {
      claimOfferData,
      isValid,
      logoUrl,
      claimThemePrimary,
      claimThemeSecondary,
    } = this.props
    const {
      claimRequestStatus,
      issuer,
      data,
      payTokenValue,
    }: ClaimOfferPayload = claimOfferData
    const logoUri = logoUrl
      ? { uri: logoUrl }
      : require('../images/cb_sovrin.png')
    const testID = 'claim-offer'
    let acceptButtonText = payTokenValue ? 'Accept & Pay' : 'Accept'
    // TODO: Get text and background color from color-picker.
    return (
      <Container style={[{ backgroundColor: claimThemePrimary }]}>
        {isValid && (
          <ClaimProofHeader
            message={`${issuer.name} is offering you`}
            title={data.name}
            onClose={this.onIgnore}
            logoUrl={logoUrl}
            testID={testID}
            payTokenValue={payTokenValue}
            containerStyle={{
              backgroundColor: claimThemePrimary,
              borderBottomColor: claimThemePrimary,
            }}
            textContainerStyle={[headerStyles.clearBg]}
            messageStyle={[headerStyles.clearBg, styles.messageStyle]}
            titleStyle={[styles.titleStyles]}
          >
            <CustomView
              fifth
              hCenter
              style={[headerStyles.headerLogoContainer, headerStyles.clearBg]}
            >
              <Icon
                absolute="TopRight"
                src={require('../images/iconClose.png')}
                medium
                testID={`${testID}-icon-close`}
                onPress={this.close}
                iconStyle={[styles.headerCloseIcon]}
                style={[styles.headerCloseIconContainer]}
              />
              <Icon
                center
                halo
                extraLarge
                resizeMode="cover"
                src={logoUri}
                style={[styles.issuerLogo]}
                iconStyle={[styles.issuerLogoIcon]}
                testID={`${testID}-issuer-logo`}
              />
            </CustomView>
          </ClaimProofHeader>
        )}
        {isValid ? (
          <ClaimOfferAttributeList list={data.revealedAttributes} />
        ) : (
          <Container fifth center>
            <CustomText h5 bg="fifth">
              Invalid claim offer. Please ignore.
            </CustomText>
          </Container>
        )}
        <FooterActions
          logoUrl={logoUrl}
          onAccept={this.onAccept}
          onDecline={this.onReject}
          denyTitle="Ignore"
          acceptTitle={acceptButtonText}
          testID={`${testID}-footer`}
          disableAccept={this.state.disableAcceptButton}
        />
        {isValid && (
          <ClaimRequestModal
            claimRequestStatus={claimRequestStatus}
            payload={claimOfferData}
            onContinue={this.close}
            senderLogoUrl={logoUrl}
            payTokenValue={payTokenValue}
            message1={payTokenValue ? 'You paid' : 'You accepted'}
            message3={payTokenValue ? '' : 'from'}
            message6={payTokenValue ? 'They will issue it to you shortly.' : ''}
            isPending={
              claimRequestStatus ===
              CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST
            }
          />
        )}
        {isValid &&
          payTokenValue && (
            <CustomModal
              buttonText="OK"
              onPress={this.hideInsufficientBalanceModal}
              onModalHide={this.onInsufficientModalHide}
              testID={`no-sufficient-balance`}
              isVisible={
                claimRequestStatus === CLAIM_REQUEST_STATUS.INSUFFICIENT_BALANCE
              }
            >
              <CustomView center doubleVerticalSpace>
                <Icon src={require('../images/alertInfo.png')} />
                <CustomText transparentBg primary center bold>
                  You do not have enough tokens to purchase this credential
                </CustomText>
              </CustomView>
            </CustomModal>
          )}
        {payTokenValue && (
          <PaymentFailureModal
            isModalVisible={
              claimRequestStatus ===
              CLAIM_REQUEST_STATUS.PAID_CREDENTIAL_REQUEST_FAIL
            }
            connectionName={issuer.name}
            testID={`${testID}-payment-failure-modal`}
            onClose={this.close}
            onRetry={this.onAccept}
          />
        )}
      </Container>
    )
  }
}

const mapStateToProps = (state: Store, props: ReactNavigation) => {
  const { claimOffer } = state
  const { uid } =
    props.navigation.state && props.navigation.state.params
      ? props.navigation.state.params
      : { uid: '' }
  const claimOfferData = claimOffer[uid]
  const logoUrl = getConnectionLogoUrl(state, claimOfferData.remotePairwiseDID)
  const themeForLogo = getConnectionTheme(state, logoUrl)
  const isValid =
    claimOfferData &&
    claimOfferData.data &&
    claimOfferData.issuer &&
    claimOfferData.data.revealedAttributes

  return {
    claimThemePrimary: themeForLogo.primary,
    claimThemeSecondary: themeForLogo.secondary,
    uid,
    claimOfferData,
    isValid,
    logoUrl,
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      claimOfferShown,
      acceptClaimOffer,
      claimOfferRejected,
      claimOfferIgnored,
      updateStatusBarTheme,
      claimOfferShowStart,
      resetClaimRequestStatus,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(ClaimOffer)

const styles = StyleSheet.create({
  headerCloseIcon: {
    marginRight: 15,
  },
  headerCloseIconContainer: {
    zIndex: 2,
  },
  strip: {
    height: 20,
    width: '100%',
    backgroundColor: color.actions.secondary,
    zIndex: -1,
  },
  issuerLogo: {
    ...StyleSheet.absoluteFillObject,
    alignSelf: 'center',
    zIndex: 1,
  },
  issuerLogoIcon: {
    borderRadius: 40,
  },
  claimOfferData: {
    zIndex: -1,
  },
  attributeListLabel: {
    flex: 4,
    paddingRight: OFFSET_3X / 2,
  },
  attributeListLabelText: {
    lineHeight: 19,
  },
  attributeListValue: {
    flex: 6,
  },
  messageStyle: {
    color: color.bg.primary.font.primary,
  },
  titleStyles: {
    backgroundColor: 'transparent',
    color: color.bg.primary.font.primary,
  },
})
