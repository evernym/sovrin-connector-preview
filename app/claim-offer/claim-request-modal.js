// @flow
import React, { PureComponent } from 'react'
import { View, StyleSheet, Image } from 'react-native'
import { Button } from 'react-native-elements'
import Modal from 'react-native-modal'
import { CustomModal, AvatarsPair, CustomText } from '../components'
import { color, font, OFFSET_1X } from '../common/styles'
import { CLAIM_REQUEST_STATUS } from './type-claim-offer'
import type { GenericObject } from '../common/type-common'
import type {
  ClaimRequestStatusModalProps,
  ClaimRequestStatusModalState,
  ClaimRequestStatus,
} from './type-claim-offer'
import { formatNumbers } from '../components/text'

// TODO: Remove the default export and use the named export
export default class ClaimRequestStatusModal extends PureComponent<
  ClaimRequestStatusModalProps,
  ClaimRequestStatusModalState
> {
  state = {
    isVisible: false,
  }

  // only states for which claim request modal will be visible
  visibleStates = [
    CLAIM_REQUEST_STATUS.SENDING_CLAIM_REQUEST,
    CLAIM_REQUEST_STATUS.CLAIM_REQUEST_SUCCESS,
    CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST,
    CLAIM_REQUEST_STATUS.PAID_CREDENTIAL_REQUEST_SUCCESS,
  ]

  onContinue = () => {
    this.hideModal()
    this.props.onContinue()
  }

  showModal() {
    this.setState({ isVisible: true })
  }

  hideModal() {
    this.setState({ isVisible: false })
  }

  toggleModal(claimRequestStatus: ClaimRequestStatus) {
    if (this.visibleStates.indexOf(claimRequestStatus) > -1) {
      this.showModal()
    } else {
      this.hideModal()
    }
  }

  componentDidMount() {
    // if modal is opened when it was unmounted or freshly mounted
    this.toggleModal(this.props.claimRequestStatus)
  }

  componentWillReceiveProps(nextProps: ClaimRequestStatusModalProps) {
    // if component was not unmounted, and claim request status updates
    this.toggleModal(nextProps.claimRequestStatus)
  }

  render() {
    const {
      claimRequestStatus,
      payload: { issuer, data },
      senderLogoUrl,
      isPending = false,
      message1,
      message3,
      message5,
      message6, //message6 is for payTokenValue related usage
      payTokenValue,
    }: ClaimRequestStatusModalProps = this.props
    let message2 = data.name
    let message4 = issuer.name
    if (isPending || payTokenValue) {
      message2 = issuer.name
      message4 = `"${data.name}"`
    }

    if (payTokenValue) {
      message2 = `for ${data.name}`
      message4 = ''
    }

    const avatarRight = senderLogoUrl
      ? { uri: senderLogoUrl }
      : require('../images/cb_sovrin.png')
    const { middleImage, middleImageStyle } = isPending
      ? payTokenValue
        ? {
            middleImage: require('../images/connectArrowsRight.png'),
            middleImageStyle: styles.connectedArrowRight,
          }
        : {
            middleImage: require('../images/connectArrows.png'),
            middleImageStyle: styles.connectedArrow,
          }
      : {
          middleImage: require('../images/checkMark.png'),
          middleImageStyle: null,
        }
    const conditionalMessage =
      message6 !== undefined ? (payTokenValue ? message6 : null) : message5
    return (
      <CustomModal
        onPress={this.onContinue}
        buttonText="Continue"
        testID={'claim-request'}
        isVisible={this.state.isVisible}
        disabled={
          claimRequestStatus ===
          CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST
        }
      >
        <AvatarsPair
          middleImage={middleImage}
          middleImageStyle={middleImageStyle}
          avatarRight={avatarRight}
          testID={'claim-request'}
        />
        <CustomText
          h5
          center
          tertiary
          bg="tertiary"
          transparentBg
          style={[styles.message]}
          testID={`claim-request-message`}
        >
          {claimRequestStatus ===
          CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST
            ? ''
            : message1}
        </CustomText>
        {payTokenValue &&
        claimRequestStatus !==
          CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST ? (
          <CustomText
            h5
            bold
            center
            tertiary
            bg="tertiary"
            transparentBg
            style={[styles.message]}
            testID={`claim-request-message`}
          >
            {`${formatNumbers(payTokenValue)} tokens`}
          </CustomText>
        ) : null}
        <CustomText
          h5
          bold
          center
          tertiary
          bg="tertiary"
          transparentBg
          style={[styles.message]}
          testID={`claim-request-message`}
        >
          {claimRequestStatus ===
          CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST
            ? 'Paying...'
            : message2}
        </CustomText>
        <CustomText
          h5
          center
          tertiary
          bg="tertiary"
          transparentBg
          style={[styles.message]}
          testID={`claim-request-message`}
        >
          {claimRequestStatus ===
          CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST
            ? ''
            : message3}{' '}
        </CustomText>
        <CustomText
          h5
          bold
          center
          tertiary
          bg="tertiary"
          transparentBg
          style={[styles.message]}
          testID={`claim-request-message`}
        >
          {claimRequestStatus ===
          CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST
            ? ''
            : payTokenValue ? '' : message4}
        </CustomText>

        <CustomText
          h5
          center
          tertiary
          bg="tertiary"
          transparentBg
          style={[styles.message]}
          testID={`claim-request-message`}
        >
          {claimRequestStatus ===
          CLAIM_REQUEST_STATUS.SENDING_PAID_CREDENTIAL_REQUEST
            ? ''
            : conditionalMessage}
        </CustomText>
      </CustomModal>
    )
  }
}

const styles = StyleSheet.create({
  connectedArrow: {
    height: 20,
    width: 80,
    zIndex: -1,
    right: 7,
  },
  connectedArrowRight: {
    height: 20,
    width: 80,
    zIndex: -1,
  },
  message: {
    marginBottom: OFFSET_1X / 2,
  },
  title: {
    marginBottom: OFFSET_1X,
  },
})
