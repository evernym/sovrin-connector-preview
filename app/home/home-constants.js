// @flow

export const FEEDBACK_TEST_ID = 'home-feedback-id-label'
export const SOVRINTOKEN_TEST_ID = 'home-sovrintoken-id-label'
export const SOVRINTOKEN_AMOUNT_TEST_ID = 'home-sovrintoken-amount'
