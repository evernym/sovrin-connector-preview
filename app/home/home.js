// @flow
import React, { PureComponent } from 'react'
import { Animated, StyleSheet, Platform } from 'react-native'
import { connect } from 'react-redux'
import {
  Container,
  CustomView,
  Icon,
  UserAvatar,
  CustomText,
  CustomHeader,
} from '../components'
import CustomActivityIndicator from '../components/custom-activity-indicator/custom-activity-indicator'
import { createStackNavigator } from 'react-navigation'
import Bubbles from './bubbles'
import {
  color,
  barStyleDark,
  OFFSET_3X,
  OFFSET_2X,
  whiteSmokeSecondary,
} from '../common/styles'
import { primaryHeaderStyles } from '../components/layout/header-styles'
import { homeRoute, walletRoute } from '../common'
import { getConnections } from '../store/connections-store'
import type { Store } from '../store/type-store'
import type { HomeProps, HomeState } from './type-home'
import {
  FEEDBACK_TEST_ID,
  SOVRINTOKEN_TEST_ID,
  SOVRINTOKEN_AMOUNT_TEST_ID,
} from './home-constants'
import { Apptentive } from 'apptentive-react-native'
import WalletBalance from '../wallet/wallet-balance'
import type { Connection } from '../store/type-connection-store'
import Banner from '../components/banner/banner'
import { NavigationActions } from 'react-navigation'
import { getUnseenMessages } from '../store/store-selector'

export class DashboardScreen extends PureComponent<HomeProps, HomeState> {
  state = {
    scrollY: new Animated.Value(0),
  }

  static navigationOptions = ({ navigation }) => ({
    header: (
      <CustomHeader backgroundColor={whiteSmokeSecondary} largeHeader>
        <WalletBalance
          render={balance => (
            <CustomView
              horizontalSpace
              row
              center
              onPress={() => {
                const navigateAction = NavigationActions.navigate({
                  routeName: walletRoute,
                  key: walletRoute,
                })
                navigation.dispatch(navigateAction)
              }}
              testID={SOVRINTOKEN_AMOUNT_TEST_ID}
            >
              <Icon
                small
                testID={SOVRINTOKEN_TEST_ID}
                src={require('../images/sovrinTokenOrange.png')}
              />
              <CustomText
                h5
                demiBold
                center
                style={[styles.floatTokenAmount]}
                transparentBg
                testID={SOVRINTOKEN_AMOUNT_TEST_ID}
                formatNumber
              >
                {balance}
              </CustomText>
            </CustomView>
          )}
        />

        <CustomView horizontalSpace>
          <Icon
            medium
            onPress={() => Apptentive.presentMessageCenter()}
            testID={FEEDBACK_TEST_ID}
            src={require('../images/icon_feedback_grey.png')}
          />
        </CustomView>
      </CustomHeader>
    ),
  })

  render() {
    const bubblesHeight = this.state.scrollY.interpolate({
      inputRange: [0, 5],
      outputRange: [0, -5],
      extrapolate: 'clamp',
    })

    const { connections: { data, hydrated }, unSeenMessages } = this.props
    // type casting from Array<mixed> to any and then to what we need
    // because flow Array<mixed> can't be directly type casted as of now
    const connections: Connection[] = (getConnections(data): any)
    const connectionsCheck = connections && connections.length > 0

    return (
      <Container tertiary>
        <Container tertiary>
          <Banner navigation={this.props.navigation} />
          {connectionsCheck && (
            <Bubbles
              navigation={this.props.navigation}
              height={bubblesHeight}
              connections={connections}
              unSeenMessages={unSeenMessages}
            />
          )}
          {!hydrated ? <CustomActivityIndicator /> : null}
        </Container>
        <CustomView vCenter style={[styles.userAvatarContainer]}>
          <UserAvatar />
        </CustomView>
      </Container>
    )
  }
}

const mapStateToProps = (state: Store) => {
  // when ever there is change in claimOffer state and proof request state
  // getUnseenMessages selector will return updated data
  // this data is used to update red dots on connection bubbles.
  let unSeenMessages = getUnseenMessages(state)
  return {
    connections: state.connections,
    unSeenMessages,
  }
}

export default createStackNavigator({
  [homeRoute]: {
    screen: connect(mapStateToProps)(DashboardScreen),
  },
})

const styles = StyleSheet.create({
  userAvatarContainer: {
    marginVertical: Platform.OS === 'ios' ? OFFSET_3X : OFFSET_3X / 2,
  },
  floatTokenAmount: {
    paddingLeft: 10,
    color: color.actions.font.seventh,
    fontSize: 26,
  },
})
