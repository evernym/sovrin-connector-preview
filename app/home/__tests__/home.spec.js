// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { CLAIM_OFFER_STATUS } from '../../claim-offer/type-claim-offer'
import {
  claimOfferRoute,
  homeTabRoute,
  authenticationRoute,
} from '../../common'
import {
  MESSAGE_TYPE,
  PUSH_NOTIFICATION_SENT_CODE,
} from '../../api/api-constants'
import { DashboardScreen } from '../home'
import {
  getNavigation,
  getStore,
  myPairWiseConnectionDetails,
  vcxSerializedConnection,
} from '../../../__mocks__/static-data'

function props(claimOfferStatus) {
  return {
    connections: {
      data: {
        '3nj819kkjywdppuje79': {
          identifier: '3nj819kkjywdppuje79',
          name: 'Test Connection',
          senderDID: '70075yyojywdppuje79',
          senderEndpoint: '34.216.340.155:3000',
          size: 100,
          logoUrl: 'https://logourl.com/logo.png',
          vcxSerializedConnection,
          ...myPairWiseConnectionDetails,
        },
      },
    },
    navigation: getNavigation(),
    claimOfferStatus: claimOfferStatus || CLAIM_OFFER_STATUS.RECEIVED,
    route: {
      currentScreen: homeTabRoute,
    },
    pushNotification: {
      notification: null,
    },
    getUserInfo: jest.fn(),
    pushNotificationReceived: jest.fn(),
    authenticationRequestReceived: jest.fn(),
    unSeenMessages: {},
  }
}

describe('<DashboardScreen />', () => {
  const store = getStore()

  jest.useFakeTimers()

  it('should render Home and redirect user to claim offer modal', () => {
    const dashboardProps = props()
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <DashboardScreen {...dashboardProps} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
