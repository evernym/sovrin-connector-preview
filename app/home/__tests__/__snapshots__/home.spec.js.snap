// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardScreen /> should render Home and redirect user to claim offer modal 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
    tertiary={true}
  >
    <View
      style={Object {}}
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="wallet-backup-banner-label-touchable"
          accessibilityTraits={undefined}
          accessible={false}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          horizontalSpace={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          row={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d0021b",
              "flexDirection": "row",
              "paddingHorizontal": 15,
              "paddingVertical": 8,
            }
          }
          testID="wallet-backup-banner-label-touchable"
          vCenter={true}
          verticalSpace={true}
        >
          <View
            clearBg={true}
            iconStyle={
              Array [
                Object {
                  "height": 25,
                  "width": 29,
                },
                Object {
                  "tintColor": "white",
                },
              ]
            }
            src={1}
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                Array [
                  Object {
                    "height": 50,
                    "width": 50,
                  },
                  null,
                  Object {
                    "height": 25,
                    "width": 29,
                  },
                  Object {
                    "tintColor": "white",
                  },
                ]
              }
            />
          </View>
          <View
            horizontalSpace={true}
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              You have new data that is not backed up.
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  null,
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              Create another backup so you don't lose anything.
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "height": 486,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": -70,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
          "zIndex": -1,
        }
      }
    >
      <Animatable.View
        animation="zoomIn"
        delay={200}
        duration={600}
        style={
          Array [
            Object {
              "position": "absolute",
            },
            Object {
              "left": 126,
              "top": 313,
            },
          ]
        }
      >
        <View
          row={true}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "zIndex": 0,
                },
              ]
            }
          >
            <View
              accessibilityLabel="bubble-3nj819kkjywdppuje79"
              accessible={true}
              imageShadow="imageShadow"
              style={
                Object {
                  "borderRadius": 50,
                  "elevation": 3,
                  "shadowColor": "rgba(0, 0, 0, 0.25)",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 8,
                }
              }
              testID="bubble-3nj819kkjywdppuje79"
            >
              <Image
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                nativeID={undefined}
                onError={[Function]}
                onLayout={undefined}
                onLoad={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={
                  Object {
                    "uri": "https://logourl.com/logo.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 66,
                      "width": 66,
                    },
                    Object {
                      "borderRadius": 33,
                    },
                  ]
                }
                testID={undefined}
              />
            </View>
          </View>
        </View>
      </Animatable.View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        collapsable={undefined}
        source={1}
        style={
          Object {
            "height": 30,
            "tintColor": "#777777",
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginVertical": 30,
      }
    }
    vCenter={true}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="user-avatar-touchable"
      accessibilityTraits={undefined}
      accessible={false}
      clearBg={true}
      extraLarge={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      resizeMode="cover"
      round={true}
      src={
        Object {
          "uri": "/var/application/DocumentDir/user-avatar.jpeg",
        }
      }
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
      testID="user-avatar-touchable"
    >
      <Image
        accessibilityLabel="user-avatar"
        accessible={true}
        resizeMode="cover"
        source={
          Object {
            "uri": "/var/application/DocumentDir/user-avatar.jpeg",
          }
        }
        style={
          Array [
            Object {
              "height": 80,
              "width": 80,
            },
            Object {
              "borderRadius": 40,
            },
          ]
        }
        testID="user-avatar"
      />
    </View>
  </View>
</View>
`;
